front/src/applications/operationalStudies/helpers/selectTrainAndPathForProjection.ts

```diff
@@ -0,0 +1,36 @@
+import type { Dispatch, SetStateAction } from 'react';
+
+const selectTrainAndPathForProjection = (
+  trainIds: number[],
+  setSelectedTrainId: Dispatch<SetStateAction<number | undefined>>,
+  setTrainIdUsedForProjection: Dispatch<SetStateAction<number | undefined>>,
+  currentProjection?: number,
+  selectedTrainId?: number
+) => {
+  if (trainIds.length === 0) {
+    return;
+  }
+
+  if (selectedTrainId && !currentProjection) {
+    // if a selected train is given, we use it for the projection
+    if (trainIds.includes(selectedTrainId)) {
```

Comment:
Nit: these two `if`s can be merged together.

Reply:
Sounds like it's not been pushed or lost in a rebase?

Reply:
Done again :white_check_mark: